### PR TITLE
Ensure we store pusher data as text

### PR DESCRIPTION
**Pull request by richvdh**
I don't think there's any need to use canonicaljson here.

Fixes: #4475.

Discuss: should we remove the `SET bytea_output TO escape` incantation?

### CODE REVIEW

**Comment by erikjohnston** [comment]
> Discuss: should we remove the `SET bytea_output TO escape` incantation?

I'm failing to grok this a bit. What happens with existing rows?

**Comment by richvdh** [comment]
> > Discuss: should we remove the `SET bytea_output TO escape` incantation?
> 
> I'm failing to grok this a bit. What happens with existing rows?

ok, so AIUI `bytea_output` controls how postgres converts `BYTEA` values into `TEXT`. There's a couple of obvious occasions when that happens:

 * when an application sends a `bytes` object (which psycopg turns into a `BYTEA`), which then has to be cast to `TEXT`, for example so that it can be stored in a `TEXT` column. I *think* this PR fixes the only time this is happening - given #6186, I think we'd know if it was happening anywhere else.

 * when we SELECT from a `BYTEA` column: psycopg is equally happy with either encoding format (see https://github.com/psycopg/psycopg2/blob/171371da5a2cfab5cbd1455c0858212191d0d065/psycopg/typecast_binary.c#L143 and the red warning at https://www.psycopg.org/docs/usage.html#binary-adaptation).

Now... it's worth noting that neither of these cases apply when selecting *existing* data from `pushers.data`, since that is a `TEXT` column. There may already be corrupt data in that table due to inserting `bytes` with `bytea_output=hex`, but this PR doesn't change that. We could consider a migration script to fix it, but I'd argue that the impact is limited and it'll only affect a very small number of installations.

So, long story short, I *think* once this PR lands, it's safe to get rid of that incantation. It'd be nice to do so, because it means we'll stand a chance of noticing any regressions rather than waiting for a postgres connection to drop its `bytea_output` setting once in a blue moon. I'm just not quite sure I'm brave enough...

**Comment by erikjohnston** [comment]
Thanks! That makes sense

**Review by erikjohnston — APPROVED** [review]
